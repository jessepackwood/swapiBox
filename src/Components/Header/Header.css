.App-header {
  height: 150px;
  color: #feda4a;
  font-family: 'starjedi';
  margin: 100px; }

.buttons {
  display: flex;
  justify-content: space-around; }
