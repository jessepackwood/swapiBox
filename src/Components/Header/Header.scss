.App-header {
  display: flex;
  justify-content: space-around;
}

.buttons {
  display: flex;
  justify-content: space-between;
  width: 80%;
}