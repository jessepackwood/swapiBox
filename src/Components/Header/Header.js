import React from 'react';
import Button from '../Button/Button';
import PropTypes from 'prop-types';
import {fetchPeople, fetchVehicles, fetchPlanets} from '../App/fetchAPI';
import './Header.css';

const Header = ({ setDisplay, showFavorites, favoriteCount }) => {
  return (
    <div className="App-header">
      <div className='buttons'>
        <Button
          buttonText={'People'}  
          functionToFire={() => setDisplay('people', fetchPeople)} 
        />
        <Button 
          buttonText={'Planets'}
          functionToFire={() => setDisplay('planets', fetchPlanets)} 
        />
        <Button
          buttonText={'Vehicles'}
          functionToFire={() => setDisplay('vehicles', fetchVehicles)}
        />
        <Button
          buttonText={'Favorites'}
          functionToFire={showFavorites}
          count={favoriteCount}
        />
      </div>
    </div>
  );
};

Header.propTypes = {
  setDisplay: PropTypes.func,
  showFavorites: PropTypes.func,
  favoriteCount: PropTypes.number
};

export default Header;