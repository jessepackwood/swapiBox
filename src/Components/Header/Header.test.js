import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme'
import Header from './Header.js'

describe('Header tests', () => {
  let renderedHeader;

  beforeEach(() => {
    renderedHeader = shallow(<Header />);
  });

  it('should exist', () => {
    expect(renderedHeader).toBeDefined();
  })

  it('should receive props', () => {
    renderedHeader = mount(<Header favoriteCount={3} />);
    expect(renderedHeader.props().favoriteCount).toEqual(3);
  });

  it('should match the snapshot', () => {
    expect(renderedHeader).toMatchSnapshot();
  });
});