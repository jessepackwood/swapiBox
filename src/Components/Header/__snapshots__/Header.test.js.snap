// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="buttons"
>
        <Button
                buttonText="People"
                functionToFire={[Function]}
        />
        <Button
                buttonText="Planets"
                functionToFire={[Function]}
        />
        <Button
                buttonText="Vehicles"
                functionToFire={[Function]}
        />
        <Button
                buttonText="Favorites"
                count={undefined}
                functionToFire={undefined}
        />
</div>,
      "className": "App-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            buttonText="People"
            functionToFire={[Function]}
/>,
          <Button
            buttonText="Planets"
            functionToFire={[Function]}
/>,
          <Button
            buttonText="Vehicles"
            functionToFire={[Function]}
/>,
          <Button
            buttonText="Favorites"
            count={undefined}
            functionToFire={undefined}
/>,
        ],
        "className": "buttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "People",
            "functionToFire": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Planets",
            "functionToFire": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Vehicles",
            "functionToFire": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Favorites",
            "count": undefined,
            "functionToFire": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="buttons"
>
          <Button
                    buttonText="People"
                    functionToFire={[Function]}
          />
          <Button
                    buttonText="Planets"
                    functionToFire={[Function]}
          />
          <Button
                    buttonText="Vehicles"
                    functionToFire={[Function]}
          />
          <Button
                    buttonText="Favorites"
                    count={undefined}
                    functionToFire={undefined}
          />
</div>,
        "className": "App-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              buttonText="People"
              functionToFire={[Function]}
/>,
            <Button
              buttonText="Planets"
              functionToFire={[Function]}
/>,
            <Button
              buttonText="Vehicles"
              functionToFire={[Function]}
/>,
            <Button
              buttonText="Favorites"
              count={undefined}
              functionToFire={undefined}
/>,
          ],
          "className": "buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonText": "People",
              "functionToFire": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonText": "Planets",
              "functionToFire": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonText": "Vehicles",
              "functionToFire": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonText": "Favorites",
              "count": undefined,
              "functionToFire": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
